lib/administration/ticket/include/irods/ticket_administration.hpp

```diff
@@ -0,0 +1,577 @@
+#ifndef IRODS_TICKET_ADMINISTRATION_HPP
+#define IRODS_TICKET_ADMINISTRATION_HPP
+
+#undef RxComm
+#undef NAMESPACE_IMPL
+
+#ifdef IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+#    define RxComm         RsComm
+#    define NAMESPACE_IMPL server
+
+struct RsComm;
+#else
+#    define RxComm         RcComm
+#    define NAMESPACE_IMPL client
+
+struct RcComm;
+#endif // IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+#include <string_view>
+#include <type_traits>
+#include <string>
+
+/// \brief Namespace to hold the ticket administration features
+namespace irods::experimental::administration::ticket
+{
+    /// \brief Struct tag to indicate admin privilege
+    ///
+    /// \since 4.3.1
+    inline struct admin_tag
+    {
+    } admin;
+
+    /// \brief Enumeration to indicate the ticket type
+    ///
+    /// \since 4.3.1
+    enum class ticket_type
+    {
+        read,
+        write
+    };
+
+    /// \brief Struct that holds the user name of the user that should be added/removed
+    /// 
+    /// \since 4.3.1
+    struct user_constraint
+    {
+        std::string_view value;
+    };
+
+    /// \brief  Struct that holds the name of the group that should be added/removed
+    /// 
+    /// \since 4.3.1
+    struct group_constraint
+    {
+        std::string_view value;
+    };
+
+    /// \brief  Struct that holds the name of the host that should be added/removed
+    ///
+    /// \since 4.3.1
+    struct host_constraint
+    {
+        std::string_view value;
+    };
+
+    /// \brief Struct to hold the use count constraint for the ticket
+    /// 
+    /// \since 4.3.1
+    struct use_count_constraint
+    {
+        int value = -1;
+    };
+
+    /// \brief Struct to hold the write count limit constraint for the ticket
+    ///
+    /// \since 4.3.1
+    struct n_writes_to_data_object_constraint
+    {
+        int value = -1;
+    };
+
+    /// \brief Struct to hold the write byte limit constraint for the ticket
+    ///
+    /// \since 4.3.1
+    struct n_write_bytes_constraint
+    {
+        int value = -1;
+    };
+
+    namespace NAMESPACE_IMPL
+    {
+        namespace detail
+        {
+            /// \brief A function to run the ticket operation
+            ///
+            /// \param[in] conn The communication Object
+            /// \param[in] command The string that specifies the command
+            /// \param[in] ticket_identifier The string to identify ticket
+            /// \param[in] command_modifier1 The string to modify the command
+            /// \param[in] command_modifier2 The string to modify the command
+            /// \param[in] command_modifier3 The string to modify the command
+            /// \param[in] command_modifier4 The string to modify the command
+            /// \param[in] run_as_admin A boolean that informs whether to run the operation as an admin
+            ///
+            /// \since 4.3.1
```

Comment:
Let's remove these comment lines because the function is not part of the public interface. It is an implementation detail.